python/ray/tune/schedulers/hyperband.py

```diff
@@ -292,7 +307,8 @@ def on_trial_remove(self, trial_runner: "trial_runner.TrialRunner", trial: Trial
         not finished."""
         bracket, _ = self._trial_info[trial]
         bracket.cleanup_trial(trial)
-        if not bracket.finished():
+        if not bracket.finished() and not bracket.is_being_processed:
```

Comment:
Because `self._process_bracket` is called in several places, putting the change here was the most straightforward solution to resolving the issue. LMK if this makes sense.